packages/hooks/README.md

```diff
@@ -0,0 +1,68 @@
+# Aries Framework Hooks
+
+This package exposes useful hooks that allow you to easily interact with AFJ.
+
+Everything exported from Hooks:
+
+```ts
+import
+	AgentProvider, {
+	useAgent,
+	useConnections,
+	useConnectionById,
+	useConnectionByState,
+	useCredentials,
+	useCredentialById,
+	useCredentialByState,
+	useProofs,
+	useProofById,
+	useProofByState
+} from '@aries-framework/hooks'
+```
+
+First step is to wrap your entire app in our `<AgentProvider/>`. The provider takes two props, the first is your `agentConfig` object and the second is your `genesisUrl`. The base of your app should look something like this:
+
+```ts
+import AgentProvider from "@aries-framework/hooks"
+
+const App = () => {
+	return (
+		<AgentProvider agentConfig={Your full agent config object} genesisUrl={Your genesis url} >
```

Comment:
Also, I'd rather have the `AgentProvider` take an `agent` instance as input, this gives a lot more flexibility to setup the agent to your requirements, while still leveraging this package

Reply:
Could this be a promise of an agent? The issue I think I saw here is that we need to have the context providers be initialized at run time.

Reply:
I guess that would work. Is it because you are not in an async context when passing the agent to the agent provider?

Reply:
I actually realized at like 3 in the morning last night that this was a somewhat dumb POV of mine--we've got built into the hooks package state properties for loading state--which means we just have to change the AgentProvider to be able to handle having `Agent` being empty initially while it initializes asynchronously outside the AgentProvider component. 
There might be some additional nuances to consider there, but overall I'd agree with this. @NeilSMyers we should also look at this.